connect/runtime/src/test/java/org/apache/kafka/connect/util/KafkaBasedLogTest.java

```diff
@@ -471,51 +468,42 @@ public void testProducerError() throws Exception {
 
         store.stop();
 
-        assertFalse(Whitebox.<Thread>getInternalState(store, "thread").isAlive());
+        assertFalse(store.thread.isAlive());
         assertTrue(consumer.closed());
-        PowerMock.verifyAll();
+        expectStop();
     }
 
     @Test
-    public void testReadEndOffsetsUsingAdmin() throws Exception {
+    public void testReadEndOffsetsUsingAdmin() {
         // Create a log that uses the admin supplier
         setupWithAdmin();
-        expectProducerAndConsumerCreate();
 
         Set<TopicPartition> tps = new HashSet<>(Arrays.asList(TP0, TP1));
         Map<TopicPartition, Long> endOffsets = new HashMap<>();
         endOffsets.put(TP0, 0L);
         endOffsets.put(TP1, 0L);
-        admin.retryEndOffsets(EasyMock.eq(tps), EasyMock.anyObject(), EasyMock.anyLong());
-        PowerMock.expectLastCall().andReturn(endOffsets).times(1);
-        admin.endOffsets(EasyMock.eq(tps));
-        PowerMock.expectLastCall().andReturn(endOffsets).times(1);
-
-        PowerMock.replayAll();
+        when(admin.retryEndOffsets(eq(tps), any(), anyLong())).thenReturn(endOffsets);
+        when(admin.endOffsets(eq(tps))).thenReturn(endOffsets);
```

Comment:
I think we technically lose coverage here in that we no longer verify that these methods are invoked once and only once. But that should be fine; those guarantees aren't crucial.